### ISSUE TITLE
Handling of sync and async effects

### ISSUE THREAD

**Issue by cornerman** [OutWatch/outwatch#321]
We have this one big elephant in the room, which is missing an appropriate issue: handling of async effects. There is an issue about supporting rendering of `Task`, `Future` and `IO` (#256), but before being able to adopt this, we need to fix how we currently handle `IO`.

We interpret any `IO[T]` (with an `AsVDomModifier[T]` instance) as an `EffectModifier` (https://github.com/OutWatch/outwatch/blob/master/outwatch/src/main/scala/outwatch/dom/VDomModifier.scala#L79) and later on just render it with `unsafeRunSync()` (https://github.com/OutWatch/outwatch/blob/master/outwatch/src/main/scala/outwatch/dom/helpers/DomUtils.scala#L240). But this is a problem, because not all `IO` are meant to be just run sync (we have `SyncIO` for this intention).

So why not simply convert `IO` to `Observable`, thereby making it async and be done with it? Because, this would kill performance for any construction using `Handler.create[T]`. By wrapping the handler in `IO` and then letting the handler update the dom, we get way too many updates. For example:

```scala
val myComponent: IO[VDomModifier] = Handler.create[T].map(handler => span(...))

div(
  "Hi",
  myComponent
)
```

If we would interpret `myComponent` async, we would first render `div("Hi")`. Afterwards when the IO is finished, we would update the div again with the needed lifetime hooks for subscribing to the handler. And then we would again update the div, whenever the handler emits a value. That means making `IO` async by default, would cost us one extra update per component.

I think, we need to make the default behaviour for `IO` async, because this is the only correct way. Therefore, we need to change the signature of `Handler.create[T]` from `IO[T]` to `SyncIO[T]` and/or parametrize the create function so you can wrap the handler in any effect type you like. This will break a lot of user-code. But we could maybe write some scalafix patches to at least help with the migration.

**Comment by marklister** [comment]
> This will break a lot of user-code.

We've done this before.  I think people using Outwatch pre 1.0 signed up for a moving target and are prepared to refactor their code...

**Comment by cornerman** [comment]
Fixed in #355
